app/src/main/java/com/example/bookclub/QRCodeImageAnalyzer.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bookclub;
+
+import static android.graphics.ImageFormat.YUV_420_888;
+import static android.graphics.ImageFormat.YUV_422_888;
+import static android.graphics.ImageFormat.YUV_444_888;
+
+import androidx.annotation.NonNull;
+import androidx.camera.core.ImageAnalysis;
+import androidx.camera.core.ImageProxy;
+
+import com.example.bookclub.utils.QRCodeFoundListener;
+import com.google.zxing.BinaryBitmap;
+import com.google.zxing.ChecksumException;
+import com.google.zxing.FormatException;
+import com.google.zxing.NotFoundException;
+import com.google.zxing.PlanarYUVLuminanceSource;
+import com.google.zxing.Result;
+import com.google.zxing.common.HybridBinarizer;
+import com.google.zxing.multi.qrcode.QRCodeMultiReader;
+
+import java.nio.ByteBuffer;
+
+public class QRCodeImageAnalyzer implements ImageAnalysis.Analyzer{
+    private QRCodeFoundListener mListener;
+
+    public QRCodeImageAnalyzer(QRCodeFoundListener mListener){
+        this.mListener = mListener;
+    }
+
+    @Override
+    public void analyze(@NonNull ImageProxy image) {
+        if (image.getFormat() == YUV_420_888 || image.getFormat() == YUV_422_888 || image.getFormat() == YUV_444_888){
+            ByteBuffer byteBuffer = image.getPlanes()[0].getBuffer();
+            byte[] imageData = new byte[byteBuffer.capacity()];
+            byteBuffer.get(imageData);
+
+            PlanarYUVLuminanceSource source = new PlanarYUVLuminanceSource(
+                    imageData,
+                    image.getWidth(), image.getHeight(),
+                    0, 0,
+                    image.getWidth(), image.getHeight(),
+                    false
+            );
+
+            BinaryBitmap binaryBitmap = new BinaryBitmap(new HybridBinarizer(source));
+
+            try{
+                Result result = new QRCodeMultiReader().decode(binaryBitmap);
```

Comment:
Do we have to create a new `QRCodeMultiReader` each time or can we just keep one as a field and use it multiple times?